bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/LuxtronikHeatpumpHandler.java

```diff
@@ -0,0 +1,340 @@
+import org.openhab.binding.luxtronikheatpump.internal.enums.HeatpumpCoolingOperationMode;
+import org.openhab.binding.luxtronikheatpump.internal.enums.HeatpumpOperationMode;
+import org.openhab.binding.luxtronikheatpump.internal.exceptions.InvalidChannelException;
+import org.openhab.binding.luxtronikheatpump.internal.exceptions.InvalidOperationModeException;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerCallback;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LuxtronikHeatpumpHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Stefan Giehl - Initial contribution
+ */
+@NonNullByDefault
+public class LuxtronikHeatpumpHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(LuxtronikHeatpumpHandler.class);
+    private final Set<ScheduledFuture<?>> scheduledFutures = new HashSet<>();
+    private static final int RETRY_INTERVAL = 60;
+    private boolean tiggerChannelUpdate = false;
+    private final LuxtronikTranslationProvider translationProvider;
+
+    public LuxtronikHeatpumpHandler(Thing thing, LuxtronikTranslationProvider translationProvider) {
+        super(thing);
+        this.translationProvider = translationProvider;
+    }
+
+    @Override
+    public void updateState(String channelID, State state) {
+        super.updateState(channelID, state);
+    }
+
+    @Override
+    public void updateProperty(String name, String value) {
+        super.updateProperty(name, value);
+    }
+
+    public void setStatusConnectionError() {
+        LuxtronikHeatpumpConfiguration config = getConfigAs(LuxtronikHeatpumpConfiguration.class);
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                "couldn't establish network connection [host '" + config.ipAddress + "']");
+    }
+
+    public void setStatusOnline() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getIdWithoutGroup();
+        logger.debug("Handle command '{}' for channel {}", command, channelId);
+        if (command == RefreshType.REFRESH) {
+            restartJobs();
```

Comment:
`REFRESH` is called for every channel. So if you have 20 channels you will restart the job 20 times. Refresh should be used to call updateState on the given channel with either a cached value or get the value from the device (although that might be expensive) or just ignore it, if it's handled by the `channelUpdaterJob`.

Reply:
Ah ok. Wasn't aware of that. Guess removing it is the best choice then. Will do that later.